src/FlagLocalization.cpp

```diff
@@ -0,0 +1,209 @@
+			//but then switch the obj variable for the next pass of the loop
+			if(obj == 1)
+			{
+				xsum1 += cloud.points[i-1].x;
+				ysum1 += cloud.points[i-1].y;
+				pts1 ++;
+			}
+			else
+			{
+				xsum2 += cloud.points[i-1].x;
+				ysum2 += cloud.points[i-1].y;
+				pts2 ++;
+			}
+
+			//Switching objects
+			if(obj == 1)
+			{
+				obj = 2;
+			}
+			else
+			{
+				obj = 1;
+			}
+		}
+	}
+	
+	//Averaging the points
+	avgPt_1.point.x = xsum1 / pts1;  
+	avgPt_1.point.y = ysum1 / pts1;
+	avgPt_1.point.z = 0;
+
+	//Calulating theta1 using trig.
+	theta1 = acos(avgPt_1.point.x / sqrt(avgPt_1.point.x*avgPt_1.point.x + avgPt_1.point.y*avgPt_1.point.y));
+
+	//This overwrites avgPt_1 with the true center of the flag by extending along the line formed by theta1 and the line from beacon to avg point
+	avgPt_1.point.x = ((sqrt(avgPt_1.point.x*avgPt_1.point.x + avgPt_1.point.y*avgPt_1.point.y) + (4*rad1)/(3*3.14159))*cos(theta1)); //4r/3pi is distance of centroid of semicircle from the center
+	avgPt_1.point.y = ((sqrt(avgPt_1.point.x*avgPt_1.point.x + avgPt_1.point.y*avgPt_1.point.y) + (4*rad1)/(3*3.14159))*sin(theta1));
+
+	avgPt_1.header.frame_id = "beacon_frame";	//Specifying what frame in header 
+
+	//The process is repeated for flag 2
+	avgPt_2.point.x = xsum2 / pts2;  
+	avgPt_2.point.y = ysum2 / pts2;
+	avgPt_2.point.z = 0;
+
+	theta2 = acos(avgPt_2.point.x / sqrt(avgPt_2.point.x*avgPt_2.point.x + avgPt_2.point.y*avgPt_2.point.y));
+
+	avgPt_2.point.x = ((sqrt(avgPt_2.point.x*avgPt_2.point.x + avgPt_2.point.y*avgPt_2.point.y) + (4*rad2)/(3*3.14159))*cos(theta2));
+	avgPt_2.point.y = ((sqrt(avgPt_2.point.x*avgPt_2.point.x + avgPt_2.point.y*avgPt_2.point.y) + (4*rad2)/(3*3.14159))*sin(theta2));
+
+	avgPt_2.header.frame_id = "beacon_frame";
+
+	//Assigns the new center points to new variables used in the transform
+	pnt1x = avgPt_1.point.x;
+	pnt2x = avgPt_2.point.x;
+
+	pnt1y = avgPt_1.point.y;
+	pnt2y = avgPt_2.point.y;
+
+	//Calculates yaw of the robot
+	yaw =  atan((pnt1y-pnt2y)/(pnt1x-pnt2x)); // radians
+
+	//Creates the transform between "beacon_frame" and "flag"
+	static tf2_ros::TransformBroadcaster br;
+	geometry_msgs::TransformStamped transformStamped;
+
+	transformStamped.header.stamp = ros::Time::now();
+	transformStamped.header.frame_id = "beacon_frame";
+	transformStamped.child_frame_id = "flag";
+
+	//Uses the center of flag 1 as the xyz coordinates for frame "flag"
+	transformStamped.transform.translation.x = pnt1x;
+	transformStamped.transform.translation.y = pnt1y;
+	transformStamped.transform.translation.z = 0.0;
+
+	//Generates a quaternion value for the orientation/rotation of "flag" based on RPY values (0, 0, yaw)
+	tf2::Quaternion q;
+	q.setRPY(0, 0, yaw);
+
+	//Setting the quaternion orientation of frame "flag"
+	transformStamped.transform.rotation.x = q.x();
+	transformStamped.transform.rotation.y = q.y();
+	transformStamped.transform.rotation.z = q.z();
+	transformStamped.transform.rotation.w = q.w();
```

Comment:
it looks like `transformStamped.transform.rotation = q;` would be valid syntax to replace these four lines